doc/acl/Extend-L3V6ACLs.md

```diff
@@ -0,0 +1,308 @@
+     * | MATCH_ICMP_TYPE      |              |      √       |
+     * | MATCH_ICMP_CODE      |              |      √       |
+     * |----------------------------------------------------|
+     * | MATCH_IP_PROTOCOL    |      √       |      √       |
+     * | MATCH_NEXT_HEADER    |      √       |      √       |
+     * | ---------------------------------------------------|
+     * | MATCH_L4_SRC_PORT    |      √       |      √       |
+     * | MATCH_L4_DST_PORT    |      √       |      √       |
+     * | MATCH_TCP_FLAGS      |      √       |      √       |
+     * |----------------------------------------------------|
+     */
+
+##### Pros
+
+
+-  Operator can create multiple L3 and L3V6 ACL tables. This would not be possible if we use option-A.
+- Gives the flexibility to the operator to utilize unused space in L3V6 ACL table for L3 ACL rules. 
+<p align=right>
+<img src="img/acl-merge.png" alt="New L3/L3V6 ACL Table optimization">
+</p>
+
+- If there are many v4 rules and many v6 rules, the user can continue to use separate L3 and L3V6 ACL Tables.
+
+<p align=right>
+<img src="img/acl-dont-merge.png" alt="New L3/L3V6 ACL Table optimization">
+</p>
+
+- No impact on other platforms  
+  
+##### Cons
+- If the operator decides to use the optimization, the operator needs to modify the ACL configuration, i.e., operator must modify the V4 ACL rules that need to be placed in L3V6 ACL table- the rule's ACL Table is renamed to the L3V6 ACL table.
+
+#### Option-C: Create a new ACL Table Type L3V4V6
+
+Create a new built-in table type called L3V4V6 with the following match types:
+
+
+    /*
+     * Supported Match Types in a new table type L3V4V6
+     * |-------------------------------------|
+     * |                      |              |
+     * |    Match Type        |  New L3V4V6  |
+     * |                      |              |
+     * |-------------------------------------|
+     * | MATCH_OUTER_VLAN_ID  |      √       |
+     * |-------------------------------------|
+     * | MATCH_ACL_IP_TYPE    |      √       |
+     * | MATCH_ETHER_TYPE     |      √       |
+     * |-------------------------------------|
+     * | MATCH_SRC_IPV6       |      √       |
+     * | MATCH_DST_IPV6       |      √       |
+     * | MATCH_SRC_IP         |      √       |
+     * | MATCH_DST_IP         |      √       |
+     * |-------------------------------------|
+     * | MATCH_ICMPV6_TYPE    |      √       |
+     * | MATCH_ICMPV6_CODE    |      √       |
+     * | MATCH_ICMP_TYPE      |      √       |
+     * | MATCH_ICMP_CODE      |      √       |
+     * |-------------------------------------|
+     * | MATCH_IP_PROTOCOL    |      √       |
+     * | MATCH_NEXT_HEADER    |      √       |
+     * | ------------------------------------|
+     * | MATCH_L4_SRC_PORT    |      √       |
+     * | MATCH_L4_DST_PORT    |      √       |
+     * | MATCH_TCP_FLAGS      |      √       |
+     * |-------------------------------------|
+     */
```

Comment:
I saw some limitation when testing the combined V4V6 ACL table. That is the count of fields is limited on some platform.

Reply:
Exactly. This is the reason we need to add platform specific capability checks and allow only on platforms that can support combining v4 and v6 matchfields in the same ACL table.